rosetta-source/src/main/rosetta/product-collateral-func.rosetta

```diff
@@ -19,4 +26,107 @@ func MergeEligibleCollateralCriteria: <"Java implementation merges criteria1 and
         criteria1 EligibleCollateralCriteria (1..1)
         criteria2 EligibleCollateralCriteria (1..1)
     output:
-        criteria EligibleCollateralCriteria (1..1)
\ No newline at end of file
+        criteria EligibleCollateralCriteria (1..1)
+
+        
+// collateral interest calculation parameters
+
+type DistributionAndInterestPayment: <"A class to specify the Distributions and Interest Payment provisions applicable to the collateral agreement.">
```

Comment:
The type `DistributionAndInterestPayment` should not be included in this contribution. It would be a duplicate of the `DistributionAndInterestPayment` type in the CSA model. Instead, the 'DistributionAndInterestPayment` will be modified in the CSA model to embed the `CollateralInterestParameters` attribute (to be worked on as part of a different model change).